packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js

```diff
@@ -2,52 +2,86 @@ import React from 'react';
         onPaymentSuccess,
         onPaymentError,
         resetReviewOrderButtonClicked
     } = props;
 
     const classes = mergeClasses(defaultClasses, propClasses);
 
-    const creditCard =
-        selectedPaymentMethod === 'braintree' ? (
-            <CreditCard
-                shouldSubmit={reviewOrderButtonClicked}
-                brainTreeDropinContainerId={
-                    'checkout-page-braintree-dropin-container'
-                }
-                onPaymentSuccess={onPaymentSuccess}
-                onPaymentError={onPaymentError}
-                resetShouldSubmit={resetReviewOrderButtonClicked}
-            />
-        ) : null;
+    const talonProps = usePaymentMethods({
+        ...paymentMethodOperations
+    });
+
+    const {
+        availablePaymentMethods,
+        currentSelectedPaymentMethod,
+        initialSelectedMethod,
+        isLoading
+    } = talonProps;
+
+    if (isLoading) {
+        return null;
+    }
+
+    const radios = availablePaymentMethods.map(({ code, title }) => {
+        // If we don't have an implementation for a method type, ignore it.
+        if (!Object.keys(PAYMENT_METHOD_COMPONENTS_BY_CODE).includes(code)) {
+            return;
+        }
+
+        const isSelected = currentSelectedPaymentMethod === code;
+        const Component = PAYMENT_METHOD_COMPONENTS_BY_CODE[code] || null;
+
+        return (
+            <div key={code} className={classes.payment_method}>
+                <Radio
+                    label={title}
+                    value={code}
+                    classes={{
+                        label: classes.radio_label
+                    }}
+                    checked={isSelected}
+                />
+                {isSelected && (
+                    <Component
+                        onPaymentSuccess={onPaymentSuccess}
+                        onPaymentError={onPaymentError}
+                        resetShouldSubmit={resetReviewOrderButtonClicked}
+                        setDoneEditing={setDoneEditing}
+                        shouldSubmit={reviewOrderButtonClicked}
+                    />
+                )}
+            </div>
+        );
+    });
 
     return (
         <div className={classes.root}>
-            <RadioGroup field="selectedPaymentMethod" initialValue="braintree">
-                <div className={classes.payment_method}>
-                    <Radio
-                        key={'braintree'}
-                        label={'Credit Card'}
-                        value={'braintree'}
-                        classes={{
-                            label: classes.radio_label
-                        }}
-                        checked={selectedPaymentMethod === 'braintree'}
-                    />
-                    {creditCard}
-                </div>
+            <RadioGroup
+                field="selectedPaymentMethod"
+                initialValue={initialSelectedMethod}
+            >
+                {radios}
```

Comment:
Similar to how you are rendering the radios and the related components, you can render the Free component as well. That way, instead of having PaymentInformation component and talon deal with Free as a special case, you can consider Free like any other payment method and pass the same props that you send to other payment methods. For instance: https://github.com/magento/pwa-studio/pull/2409/files#diff-c81c1a48df91716fa25273ee8d1fb34bR40

Reply:
All else aside Free _is_ a special case in that it is always implicitly the selected method on the cart if it is available. Because of that we must have special handling for it, regardless of where that logic lives. 

That said, it would be easy to add other payment methods and I think the API for adding methods will become clearer as we add less complicated ones like "Check/Money Order". It's just out of scope for this PR.